src/App.tsx

```diff
@@ -1,19 +1,153 @@
 import React from 'react';
 import './App.css';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+const preparedTodos: Todo[] = todosFromServer.map(todo => {
+  const user = usersFromServer.find(person => person.id === todo.userId) || null;
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
-};
+  return { ...todo, user };
+});
+
+interface State {
+  todos: Todo[];
+  query: string;
+  users: string;
+  userSelected: boolean;
+  titleAdded: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    todos: [...preparedTodos],
+    query: '',
+    users: '',
+    userSelected: true,
+    titleAdded: true,
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement> |
+  React.ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value,
+    } as unknown as Pick<State, keyof State>);
+
+    if (name === 'query') {
+      this.setState({
+        titleAdded: true,
+      } as Pick<State, keyof State>);
+    } else {
+      this.setState({
+        userSelected: true,
+      } as Pick<State, keyof State>);
+    }
+  };
+
+  addNewTodo = (todo: Todo) => {
+    this.state.todos.push(todo);
```

Comment:
do not use push. Use copying of an array. By the way, you don't use this method at all, so you can remove it